03.bowling/bowling.rb

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+score = ARGV[0]
+scores = score.split(',')
+shots = []
+scores.each do |s|
+  if s == 'X'
+    shots << 10
+    shots << 0
+  else
+    shots << s.to_i
+  end
+end
+
+frames = []
+shots.each_slice(2) do |s|
+  frames << s
+end
+
+point = 0
+frames.each_with_index do |frame, index|
+  if index <= 8 # 1~9フレーム
+    point += if frame[0] == 10 && frames[index + 1][0] == 10 # 連続ストライクの時
+               frame.sum + frames[index + 1][0] + frames[index + 2][0]
+             elsif frame[0] == 10 # 連続しないストライクの時
+               frame.sum + frames[index + 1][0] + frames[index + 1][1]
+             elsif frame.sum == 10 # スペアの時
+               frame.sum + frames[index + 1][0]
+             else
+               frame.sum
+             end
```

Comment:
インデントが深くて読みづらいので、こっちの方がいいかもしれません。

```suggestion
    if frame[0] == 10 && frames[index + 1][0] == 10 # 連続ストライクの時
      point += frame.sum + frames[index + 1][0] + frames[index + 2][0]
    elsif frame[0] == 10 # 連続しないストライクの時
      point += frame.sum + frames[index + 1][0] + frames[index + 1][1]
    elsif frame.sum == 10 # スペアの時
      point += frame.sum + frames[index + 1][0]
    else
      point += frame.sum
    end
```

あと、

```ruby
next_frame = frames[index + 1]
```

のように変数に入れるとコードが読みやすくなります。

Reply:
「次のフレーム」、「次の次のフレーム」を変数に格納して、条件分岐の書き方を変更しました。